arbos/arbostypes/messagewithmeta.go

```diff
@@ -32,6 +35,15 @@ var TestMessageWithMetadataAndRequestId = MessageWithMetadata{
 	Message: &TestIncomingMessageWithRequestId,
 }
 
+// IsTxTimeboosted given a tx's index in the block returns whether the tx was timeboosted or not
+func (b BlockMetadata) IsTxTimeboosted(txIndex int) bool {
```

Comment:
this could return bool, error
Then if either b == nil or len(b)  == 0 can return an error which means blockmetadata isn't set

Reply:
added this